packages/builder/src/pages/builder/app/[application]/design/[screenId]/[componentId]/_components/Component/ComponentSettingsPanel.svelte

```diff
@@ -39,6 +49,20 @@
 {#if $selectedComponent}
   {#key $selectedComponent._id}
     <Panel {title} icon={componentDefinition?.icon} borderLeft wide>
+      <span class="panel-title-content" slot="panel-title-content">
+        <input
+          class="input"
+          value={title}
+          {title}
+          placeholder={getComponentText(componentInstance)}
+          on:change={e => {
+            onUpdateName(e.target.value)
+            if (e.target === document.activeElement) {
```

Comment:
Could you explain what this does? I'm not entirely sure what checking the target (which is surely always the input component) against the document active element would do. If this is to handle an enter keypress we would be better to check if `e.key` is `"Enter"` and then blur the field.

Reply:
As you point out, I think it's just an unnecessary check.  I'll switch it out for a key press event 👍